packages/dialect-react-ui/components/Subscribe/index.tsx

```diff
@@ -0,0 +1,279 @@
+              isSubscribed={false}
+              isLoading={isSigningMessage}
+              onSubscribe={() => {
+                onWalletConnect();
+                setAutoSubscribe(true);
+              }}
+            />
+          );
+        }
+        return (
+          <ConnectedSubscribe
+            autoSubscribe={autoSubscribe}
+            onOpenMoreOptions={openModal}
+          />
+        );
+      }}
+    </WalletStatesWrapper>
+  );
+}
+
+interface ConnectedSubscribeProps {
+  autoSubscribe?: boolean;
+  onOpenMoreOptions: () => void;
+  onSubscribe?: (thread: Thread) => void;
+}
+
+function ConnectedSubscribe({
+  autoSubscribe,
+  onOpenMoreOptions,
+  onSubscribe,
+}: ConnectedSubscribeProps): JSX.Element {
+  const { adapter: wallet } = useDialectWallet();
+
+  const { dappAddress } = useDialectDapp();
+  if (!dappAddress) {
+    throw new Error(
+      'dapp address should be provided for subscribe button to work'
+    );
+  }
+
+  const { create: createThread, isCreatingThread } = useThreads();
+
+  const [isAutoSubscribed, setAutoSubscribed] = useState(false);
+
+  const {
+    globalAddress: walletAddress,
+    create: createAddress,
+    isCreatingAddress,
+    isDeletingAddress,
+  } = useNotificationChannel({ addressType });
+
+  const {
+    enabled: subscriptionEnabled,
+    toggleSubscription,
+    isToggling,
+    isFetchingSubscriptions,
+  } = useNotificationChannelDappSubscription({
+    addressType,
+  });
+
+  const {
+    connected: {
+      solana: { shouldConnect: isSolanaShouldConnect },
+      dialectCloud: { shouldConnect: isDialectCloudShouldConnect },
+    },
+  } = useDialectConnectionInfo();
+
+  const isBackendSelectable =
+    isSolanaShouldConnect && isDialectCloudShouldConnect;
+
+  const backend =
+    isSolanaShouldConnect && !isBackendSelectable
+      ? Backend.Solana
+      : Backend.DialectCloud;
+
+  const { thread, isFetchingThread } = useThread({
+    findParams: { otherMembers: dappAddress ? [dappAddress] : [] },
+  });
+
+  const createWalletThread = useCallback(async () => {
+    if (!dappAddress) return;
+    return createThread({
+      me: { scopes: [ThreadMemberScope.ADMIN] },
+      otherMembers: [
+        { publicKey: dappAddress, scopes: [ThreadMemberScope.WRITE] },
+      ],
+      encrypted: false,
+      backend,
+    });
+  }, [backend, createThread, dappAddress]);
+
+  const createWalletAddress = useCallback(async () => {
+    if (!wallet.publicKey) {
+      return;
+    }
+    return createAddress({ value: wallet.publicKey?.toBase58() });
+  }, [createAddress, wallet.publicKey]);
+
+  const fullEnableWallet = useCallback(async () => {
+    const address = await createWalletAddress();
+    const thread = await createWalletThread();
+    if (!thread) {
+      return;
+    }
+    await toggleSubscription({ enabled: true, address });
+    onSubscribe?.(thread);
+  }, [
+    createWalletAddress,
+    createWalletThread,
+    onSubscribe,
+    toggleSubscription,
+  ]);
```

Comment:
can this logic be reused? guess the same code we have in wallet settings component

Reply:
Yep, but I follow the rule of a thumb — it's to early to re-use logic if it's only two places — once we have more usage — could refactor or as it gets mature.